kvclientbabysitter.go

```diff
@@ -0,0 +1,559 @@
+package gocbcorex
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/couchbase/gocbcorex/memdx"
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"golang.org/x/exp/slices"
+)
+
+type KvTargetTlsConfig struct {
+	RootCAs            *x509.CertPool
+	InsecureSkipVerify bool
+	CipherSuites       []uint16
+}
+
+func (v KvTargetTlsConfig) Equals(o *KvTargetTlsConfig) bool {
+	return v.RootCAs.Equal(o.RootCAs) &&
+		v.InsecureSkipVerify == o.InsecureSkipVerify &&
+		slices.Equal(v.CipherSuites, o.CipherSuites)
+}
+
+type KvTarget struct {
+	Address   string
+	TLSConfig *KvTargetTlsConfig
+}
+
+func (v KvTarget) Equals(o KvTarget) bool {
+	if v.Address != o.Address {
+		return false
+	}
+
+	if v.TLSConfig == nil && o.TLSConfig == nil {
+		// both nil, so equal
+	} else if v.TLSConfig != nil && o.TLSConfig != nil {
+		if !v.TLSConfig.Equals(o.TLSConfig) {
```

Comment:
```suggestion
		return v.TLSConfig.Equals(o.TLSConfig)
```

Reply:
I intentionally opted to do it the original way to make it more obvious that we are testing for all the conditions where it is false, and then only returning true if everything matches.  This structure will help if there are additional checks added in the future.